src/fdcache_page.cpp

```diff
@@ -774,7 +774,7 @@ bool PageList::GetNoDataPageLists(fdpage_list_t& nodata_pages, off_t start, size
 
         fdpage  tmppage;
         tmppage.offset   = std::max(iter->offset, start);
-        tmppage.bytes    = (-1 != stop_pos ? iter->bytes : std::min(iter->bytes, (stop_pos - tmppage.offset)));
+        tmppage.bytes    = (-1 == stop_pos ? iter->bytes : std::min(iter->bytes, (stop_pos - tmppage.offset)));
```

Comment:
@gaul 
@leon-inf's fix is correct.
I think this is a result of writing the wrong code by me.

I think that this mis-coding probably didn't make any effect.
It may have failed by trying to open a hole larger than the file size when creating a PuchHole in the file.
And the caller of this function doesn't care about the errors and treats them as warnings.
In other words, the result is that it was not possible to create a hole in the file.
(It is possible that the disk free space could not be wide. I remember that this may have been reported in another issue.)

I think inspecting this code is difficult to do from the outside(ex. test script).
So I think this PR is best merged as-is.